### PR TITLE
Add functional Talis scraper

### PR DESCRIPTION
**Pull request by HarrisonGregg**

### CODE REVIEW

File: osp_scraper/spiders/talis.py [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+        "http://readinglist.bi.edu/index.html",
+        "https://auckland.rl.talis.com/index.html",
+        "https://victoria.rl.talis.com/index.html",
+        "https://waikato.rl.talis.com/index.html"
+    ]
+
+    def start_requests(self):
+        for url in self.database_urls:
+            yield scrapy.Request(
+                urljoin(url, "/search.html?q=a"),
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/talis.py]: Oh, good idea.

File: osp_scraper/spiders/talis.py [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                    'depth': 0,
+                    'hops_from_seed': 0
+                },
+                callback=self.parse_for_courses
+            )
+
+    def parse_for_courses(self, response):
+        for a_tag in response.css("#search_results_lists ol a"):
+            anchor = a_tag.css("::text").extract_first()
+            url = a_tag.css("::attr(href)").extract_first()
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/talis.py]: Oh right.

File: osp_scraper/spiders/talis.py [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+            yield response.follow(
+                url,
```
Comment (HarrisonGregg) [inline osp_scraper/spiders/talis.py]: Cool, I didn't realize.  Seems to work.

**Comment by HarrisonGregg** [comment]
@jodizzle Good ideas.  I made all the recommended changes.

File: osp_scraper/spiders/talis.py [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        "http://readinglist.bi.edu/index.html",
+        "https://auckland.rl.talis.com/index.html",
+        "https://victoria.rl.talis.com/index.html",
+        "https://waikato.rl.talis.com/index.html"
+    ]
+
+    def start_requests(self):
+        for url in self.database_urls:
+            yield scrapy.Request(
+                urljoin(url, "/search.html?q=+"),
```
Comment (jodizzle) [inline osp_scraper/spiders/talis.py]: It'd be nice to have a comment that describes what this line does/why you're searching like that.

File: osp_scraper/spiders/talis.py [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+                    'source_url': response.url,
+                    'source_anchor': anchor
+                },
+                callback=self.parse_for_courses
+            )
+
+    def extract_links(self, response):
+        title = response.css("#pageTitle::text").get()
+        url = response.css("#exportsToRIS::attr(href)").get()
+        yield (url, title)
```
Comment (jodizzle) [inline osp_scraper/spiders/talis.py]: Newline?

**Pull request merged**